subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -111,13 +114,14 @@ static void state_machine_button_process(
 	switch (action) {
 	case BT_ENOCEAN_BUTTON_PRESS:
 		button->state = BT_MESH_SILVAIR_ENOCEAN_STATE_WAIT;
+		button->start_ts = k_uptime_get();
 		button->target = on_off;
 		k_work_reschedule(&button->timer,
 			K_MSEC(CONFIG_BT_MESH_SILVAIR_ENOCEAN_WAIT_TIME));
 		break;
 	case BT_ENOCEAN_BUTTON_RELEASE:
 		if (button->state != BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA ||
-		    !k_work_delayable_is_pending(&button->timer) ||
+		    !button->onoff_disabled ||
```

Comment:
I might be missing something, but `onoff_disabled` is set to `true` only one place, and `button->state` is set to `BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA` in that same place. Why do you need `onoff_disabled`?

Reply:
You actually pointed out something here, but the reason why we need the onoff_disabled is in the case where the button has been pressed for over 6 seconds. In that instance the state will still be `BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA` but we want to send a onoff anyways. We can however remove the `(button->state != BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA` check, since the onoff_disabled covers all instances.